src/cartservice/Program.cs

```diff
@@ -104,55 +112,27 @@ static void Main(string[] args)
-                            {
-                                Console.WriteLine($"Reading cart service port from {CART_SERVICE_PORT} environment variable");
-                                string portStr = Environment.GetEnvironmentVariable(CART_SERVICE_PORT);
-                                if (string.IsNullOrEmpty(portStr))
-                                {
-                                    Console.WriteLine($"{CART_SERVICE_PORT} environment variable was not set. Setting the port to 8080");
-                                    port = 8080;
-                                }
-                                else    
-                                {
-                                    port = int.Parse(portStr);
-                                }
-                            }
+                            int port = ReadParameter("cart service port", options.Port, CART_SERVICE_PORT, int.Parse, 8080);
 
-                            // Set redis cache host (hostname+port)
-                            ICartStore cartStore;
-                            string redis = ReadRedisAddress(options.Redis);
+                            string projectId = ReadParameter("cloud service project id", options.ProjectId, PROJECT_ID, p => p, null);
 
-                            // Redis was specified via command line or environment variable
-                            if (!string.IsNullOrEmpty(redis))
+                            // Initialize Stackdriver Exporter - currently for tracing only
+                            if (!string.IsNullOrEmpty(projectId))
                             {
-                                // If you want to start cart store using local cache in process, you can replace the following line with this:
-                                // cartStore = new LocalCartStore();
-                                cartStore = new RedisCartStore(redis);
-
-                                return StartServer(hostname, port, cartStore);
-                            }
-                            else
-                            {
-                                Console.WriteLine("Redis cache host(hostname+port) was not specified. Starting a cart service using local store");
-                                Console.WriteLine("If you wanted to use Redis Cache as a backup store, you should provide its address via command line or REDIS_ADDRESS environment variable.");
-                                cartStore = new LocalCartStore();
+                                var exporter = new StackdriverExporter(
+                                    projectId, 
+                                    Tracing.ExportComponent,
+                                    viewManager: null);
+                                exporter.Start();
                             }
 
+                            // Set redis cache host (hostname+port)
+                            string redis = ReadParameter("redis cache address", options.Redis, REDIS_ADDRESS, p => p, null);
```

Comment:
why don't you put all the ReadParameter stuff next to each other (initialization) ––and then start connecting to things like stackdriver or redis?